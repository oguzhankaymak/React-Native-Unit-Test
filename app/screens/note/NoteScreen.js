import React, {useState} from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import AddNoteFormModal from '../../components/addNoteFormModal/AddNoteFormModal';
import NoteCard from '../../components/noteCard/NoteCard';

import styles from './styles/NoteScreenStyle';

const NoteScreen = () => {
  const [modalVisible, setModalVisible] = useState(false);
  const [noteList, setNoteList] = useState([]);

  const addNote = text => {
    if (text && text.length > 0) {
      let tempValue = {
        id: Math.floor(Math.random() * 100000),
        text,
      };

      setNoteList([...noteList, tempValue]);
    }
  };

  const deleteNote = id => {
    let newNotes = noteList.filter(data => data.id !== id);
    setNoteList(newNotes);
  };

  const _renderContent = () => {
    if (noteList && noteList.length > 0) {
      return noteList.map(note => (
        <View key={note.id} style={styles.noteCardView}>
          <NoteCard note={note} deleteNote={id => deleteNote(id)} />
        </View>
      ));
    }
    return (
      <View>
        <Text>Note Not Added Yet</Text>
      </View>
    );
  };

  const openModal = () => setModalVisible(true);
  const closeModal = () => setModalVisible(false);

  return (
    <View style={styles.container}>
      {modalVisible && (
        <AddNoteFormModal closeModal={closeModal} addNote={addNote} />
      )}

      <View style={styles.header}>
        <Text style={styles.headerTitleText}>Note Screen</Text>
        <TouchableOpacity style={styles.addButton} onPress={openModal}>
          <Text style={styles.addButtonText}>Add Note</Text>
        </TouchableOpacity>
      </View>

      <View style={styles.body}>
        <Text style={styles.contentTitleText}>Note List</Text>
        <View style={styles.line} />
        <View style={styles.content} testID={'note-content'}>
          {_renderContent()}
        </View>
      </View>
    </View>
  );
};

export default NoteScreen;
