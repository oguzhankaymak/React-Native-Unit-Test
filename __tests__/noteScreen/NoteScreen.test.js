import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import NoteScreen from '../../app/screens/note/NoteScreen';

test("if it hasn't note, displaying description message ", () => {
  const noNoteMessage = 'Note Not Added Yet';

  const {getByText} = render(<NoteScreen />);
  expect(getByText(noNoteMessage)).toBeTruthy();
});

test('when new notes added, these notes are displaying', () => {
  const note_1 = 'Blog yazımı tamamlamam gerekiyor.';
  const note_2 = 'Alışveriş yapmam gerekiyor.';

  const {getByText, getByPlaceholderText, getByTestId, toJSON} = render(
    <NoteScreen />,
  );

  //open modal
  const openModalButton = getByText('Add Note');
  fireEvent.press(openModalButton);

  const input = getByPlaceholderText('Write note...');
  const addNotebutton = getByText('Add');

  //note 1
  fireEvent.changeText(input, note_1);
  fireEvent.press(addNotebutton);

  //note 2
  fireEvent.changeText(input, note_2);
  fireEvent.press(addNotebutton);

  // note count control
  const noteContent = getByTestId('note-content');
  expect(noteContent.children.length).toBe(2);

  expect(getByText(note_1)).toBeTruthy();
  expect(getByText(note_2)).toBeTruthy();
  expect(toJSON()).toMatchSnapshot();
});

test("when deleted note, this note don't display", () => {
  const note_1 = 'Blog yazımı tamamlamam gerekiyor.';
  const note_2 = 'Alışveriş yapmam gerekiyor.';

  const {getByText, getAllByText, getByPlaceholderText, getByTestId} = render(
    <NoteScreen />,
  );

  const openModalButton = getByText('Add Note');
  fireEvent.press(openModalButton);

  const input = getByPlaceholderText('Write note...');
  const addNotebutton = getByText('Add');

  //note 1
  fireEvent.changeText(input, note_1);
  fireEvent.press(addNotebutton);

  //note 2
  fireEvent.changeText(input, note_2);
  fireEvent.press(addNotebutton);

  //delete note
  const deletebutton = getAllByText('delete');
  fireEvent.press(deletebutton[0]);

  const noteContent = getByTestId('note-content');
  expect(noteContent.children.length).toBe(1);
});
