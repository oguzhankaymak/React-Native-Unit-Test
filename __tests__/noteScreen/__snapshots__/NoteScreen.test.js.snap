// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notes showing when notes added 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#18FDFA80",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "elevation": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
            "width": 250,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "padding": 2,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 30,
                "left": 30,
                "right": 30,
                "top": 30,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "borderRadius": 10,
                  "height": 24,
                  "justifyContent": "center",
                  "width": 24,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                  }
                }
              >
                X
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingBottom": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Add Note
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Write note..."
            rejectResponderTermination={true}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#A9A9A9",
                "borderRadius": 10,
                "borderWidth": 1,
                "fontSize": 13,
                "height": 35,
                "marginTop": 15,
                "paddingVertical": 0,
                "width": 200,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4BB543",
                "borderRadius": 20,
                "height": 30,
                "justifyContent": "center",
                "marginTop": 15,
                "opacity": 1,
                "width": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 13,
                }
              }
            >
              Add
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 40,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
        }
      }
    >
      Note Screen
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4BB543",
          "borderRadius": 20,
          "height": 30,
          "justifyContent": "center",
          "marginTop": 15,
          "opacity": 1,
          "width": 720,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 15,
          }
        }
      >
        Add Note
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Note List
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#A9A9A9",
          "height": 0.8,
          "marginTop": 8,
          "width": 720,
        }
      }
    />
    <View
      style={
        Object {
          "paddingVertical": 15,
        }
      }
      testID="note-content"
    >
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#C0C0C0",
              "borderRadius": 8,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 8,
            }
          }
        >
          <View>
            <Text>
              Blog yazımı tamamlamam gerekiyor.
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#d9534f",
                "borderRadius": 5,
                "height": 34,
                "justifyContent": "center",
                "opacity": 1,
                "width": 50,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                }
              }
            >
              delete
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#C0C0C0",
              "borderRadius": 8,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 8,
            }
          }
        >
          <View>
            <Text>
              Alışveriş yapmam gerekiyor.
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#d9534f",
                "borderRadius": 5,
                "height": 34,
                "justifyContent": "center",
                "opacity": 1,
                "width": 50,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                }
              }
            >
              delete
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
